docs/source/features/parallelisms.rst

```diff
@@ -80,10 +72,35 @@ The distributed optimizer in NeMo is built on top of
 `DistributedFusedAdam <https://github.com/NVIDIA/apex/blob/master/apex/contrib/optimizers/distributed_fused_adam.py>`_
 from Apex.
 
+Fully-Shared Data Parallelism (FSDP)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+NeMo supports Fully-Sharded Data Parallelism (FSDP) that shards parameter gradients and low-precision parameters for computation on top of the model states that Distributed optimizer shards (optimizer states and high-precision parameters).
+Since FSDP shards the entire model states, it ensures linear model state memory saving with increasing DP size.
+FSDP can be preferred for the LLM training with unbalanced workload between pipeline stages (or Transformer layers) or with a large vocabulary size, where pipelining would cause huge computation bubbles due to the workload imbalance.
+
+NeMo uses `pytorch's FSDP interface <https://pytorch.org/tutorials/intermediate/FSDP_tutorial.html>`_ to shard LLM model states, which flattens the parameters of each Transformer layer and partitions across datap-parallel GPUs.
+FSDP introduces collectives across data-parallel GPUs; all-gather of the parameters for computation and reduce-scatter of parameter gradients.
+The parameter-all gather happens in both network forward- and back-propagation phases, and the gradient reduce-scatter happens only in the back-propagation.
+These FSDP communications are overlapped with Transformer layer computations.
+
+Setting ``fsdp=true`` enables FSDP.
+The mixed precision recipe can be set by ``precision`` knob, which determines both the computation and communication precisions.
+Also, one can use ``grad_reduce_dtype`` to override the gradient reduction precision specifically.
+
```

Comment:
can we add one or two sentences comparing FSDP vs.  Mcore MP(TP/PP/CP) w/ regard to performance and our suggestions in which case to use fsdp or mcore mp?

Reply:
I wonder if we can specify how one can used FSDP+TP.

Reply:
@yaoyu-33 
I think the below line explains that? 
https://github.com/NVIDIA/NeMo/blob/56493133d8c049abdd7ce699618574d5d6da6b4a/docs/source/features/parallelisms.rst?plain=1#L80

Let me add one more sentence regarding the preference. However, I would rather not give too much specifics given the current torch FSDP has inefficiency and I don't want to give a wrong impression that FSDP is fundamentally worse than 3D parallelism.

Reply:
@ntajbakhsh 
Yes, I will add that.